include/fiction/algorithms/simulation/sidb/detect_bdl_wires.hpp

```diff
@@ -597,11 +597,13 @@ class detect_bdl_wires_impl
      * lengths are found.
      *
      * @param type The type of the cell to filter by.
+     * @param forbidden_type The type of cell that is forbidden in the wire and is automatically filtered out.
      * @return A vector of `bdl_wire` objects containing cells of the specified type. If no such wires are found,
      *         an empty vector is returned.
      */
     [[nodiscard]] std::vector<bdl_wire<Lyt>>
-    filter_wires_by_type(const typename technology<Lyt>::cell_type& type) const noexcept
+    filter_wires_by_type(const typename technology<Lyt>::cell_type& type,
+                         const typename technology<Lyt>::cell_type  forbidden_type) const noexcept
```

Comment:
Maybe I'm missing something here. If you filter wires by a certain type, why do you need to specify another type that is prohibited? Wouldn't any type `!= type` be automatically prohibited by definition?

Reply:
Yes, I understand that this is not great and confusing. Thanks for bringing this up!
I think we should solve this differently and use a different name for the input. The idea is that ``forbidden_type`` can be used to remove BDL pairs of the given type from the wire. This is needed in the specific case of a BDL wire with input and output BDL pairs (typically the BDL wires of our gates have either input or output BDL wires and not both).

Reply:
hmm, what do you need about the new version.

Reply:
I like it better. Thank you!